go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -557,19 +557,27 @@ func (tsv *TabletServer) setTimeBomb() chan struct{} {
 	return done
 }
 
-// IsHealthy returns nil if the query service is healthy (able to
-// connect to the database and serving traffic) or an error explaining
+// IsHealthy returns nil for non-serving types or if the query service is healthy (able to
+// connect to the database and serving traffic), or an error explaining
 // the unhealthiness otherwise.
 func (tsv *TabletServer) IsHealthy() error {
-	_, err := tsv.Execute(
-		tabletenv.LocalContext(),
-		nil,
-		"select 1 from dual",
-		nil,
-		0,
-		nil,
-	)
-	return err
+	tsv.mu.Lock()
+	tabletType := tsv.target.TabletType
+	tsv.mu.Unlock()
+	switch tabletType {
+	case topodatapb.TabletType_MASTER, topodatapb.TabletType_REPLICA, topodatapb.TabletType_BATCH:
```

Comment:
To be future-proof, let's add `experimental`, which is considered a serving type.

Reply:
Oh. And rdonly also.

Reply:
BATCH and RDONLY are the same value -- the switch complains if i add that.  I'll do experimintal tho

Reply:
Of course :). Just experimental then.